### PR TITLE
Add ulfius_websocket_send_json_message

### PR DESCRIPTION
**Pull request by oliv3**

### CODE REVIEW

File: include/ulfius.h [hunk 1]
```diff
@@ -1553,6 +1553,17 @@ int ulfius_websocket_send_fragmented_message(struct _websocket_manager * websock
...
 
+/**
+ * Sends a JSON message in the websocket
+ * @param websocket_manager the websocket manager to use for sending the message
+ * @param message the message to send
+ * @return U_OK on success
+ */
+#ifndef U_DISABLE_JANSSON
+int ulfius_websocket_send_json_message(struct _websocket_manager * websocket_manager,
+                                       const json_t * message);
```
Comment (babelouest) [inline include/ulfius.h:1564]: Better off leaving the `const` keyword. Sometimes the compiler is unhappy when you pass a `json_t *` and it expects a `const json_t *`...
Also, if it is a const pointer, only the pointer would be a constant, not its content

File: src/u_websocket.c [hunk 2]
```diff
@@ -1150,6 +1150,20 @@ int ulfius_websocket_send_message(struct _websocket_manager * websocket_manager,
...
+/**
+ * Send a JSON message in the websocket
+ * Return U_OK on success
+ */
+#ifndef U_DISABLE_JANSSON
+int ulfius_websocket_send_json_message(struct _websocket_manager * websocket_manager,
+                                       const json_t *message) {
+  char * json = json_dumps(message, JSON_COMPACT);
+  int ret = ulfius_websocket_send_message(websocket_manager, U_WEBSOCKET_OPCODE_TEXT, strlen(json), json);
+  free(json);
```
Comment (babelouest) [inline src/u_websocket.c:1162]: Stricto senso this is correct, but in ulfius' context it's better to use `o_malloc`/`o_free`, for when a program uses a different memory allocation system.

File: include/ulfius.h (hunk 1 above)
Comment (oliv3) [reply include/ulfius.h:1564]: Fixed.

File: src/u_websocket.c (hunk 2 above)
Comment (oliv3) [reply src/u_websocket.c:1162]: Fixed.

File: API.md [hunk 3]
```diff
@@ -1233,6 +1233,12 @@ int ulfius_websocket_send_fragmented_message(struct _websocket_manager * websock
                                              const uint64_t data_len,
                                              const char * data,
                                              const size_t fragment_len);
+/**
+ * Send a JSON message in the websocket
+ * Return U_OK on success
+ */
+int ulfius_websocket_send_json_message(struct _websocket_manager * websocket_manager,
+                                       const json_t * message);
```
Comment (babelouest) [inline API.md:1241]: remove `const` there too please :)

Comment (oliv3) [reply API.md]: Oups, fixed.

File: src/u_websocket.c (hunk 2 above)
Comment (oliv3) [reply src/u_websocket.c]: Right, I missed the call to `json_set_alloc_funcs()`.

**Comment by babelouest** [comment]
Thanks @oliv3 !

**Pull request merged**